lib/discourse_api/single_sign_on.rb

```diff
@@ -99,11 +99,7 @@ def unsigned_payload
        payload[k] = val
       end
 
-      if @custom_fields
-        @custom_fields.each do |k, v|
-          payload["custom.#{k}"] = v.to_s
-        end
-      end
+      @custom_fields.each { |k, v| payload[k] = v.to_s } if @custom_fields
```

Comment:
Sorry for changing my mind on this, but I think we should just maintain backwards compatibility here. Let's check if it starts with `custom.` and if not add it. I still don't understand why people would use the single_sign_on class directly over api/sso but let's just not break things here.

Reply:
Will do in the next few days 👍

Reply:
Is this still something you need/want to work on?

Reply:
Hey, yea I'll do it this week. Sorry for the delay, moved to a non-discourse project at work.

Reply:
Next week 😬 

I'm in the process of updating the gems on my project to upgrade to Rails 6. Going to open a separate PR to update to Faraday 1.0 as well. Got a few other gem issues I have to get through first before circling back to this.